graylog2-server/src/main/java/org/graylog2/rest/resources/users/UsersResource.java

```diff
@@ -233,15 +234,27 @@ private UserSummary returnSummary(UserContext userContext, User user) {
      */
     @GET
     @Deprecated
-    @RequiresPermissions(RestPermissions.USERS_LIST)
     @ApiOperation(value = "List all users", notes = "Permissions and session data included by default")
     public UserList listUsers(
             @ApiParam(name = "include_permissions") @QueryParam("include_permissions") @DefaultValue("true") boolean includePermissions,
-            @ApiParam(name = "include_sessions") @QueryParam("include_sessions") @DefaultValue("true") boolean includeSessions) {
-        return listUsersSelective(includePermissions, includeSessions);
+            @ApiParam(name = "include_sessions") @QueryParam("include_sessions") @DefaultValue("true") boolean includeSessions,
+            @Context SearchUser searchUser) {
+        return searchUser.isPermitted(RestPermissions.USERS_LIST) ?
+                listUsersSelective(includePermissions, includeSessions) :
+                listForLoggedInUser(searchUser, includePermissions, includeSessions);
+    }
+
+    private UserList listForLoggedInUser(final SearchUser searchUser, final boolean includePermissions, final boolean includeSessions) {
+        final var user = userManagementService.load(searchUser.username());
```

Comment:
Is there a reason why you are not just using `searchUser.getUser()` here?

Reply:
no, fixed it